grails-app/controllers/org/pih/warehouse/inventory/InventoryController.groovy

```diff
@@ -574,13 +575,12 @@ class InventoryController {
     }
 
 
-    def listExpiredStock() {
-        def location = Location.get(session.warehouse.id)
-        def categorySelected = (params.category) ? Category.get(params.category) : null
-        def inventoryItems = dashboardService.getExpiredStock(categorySelected, location)
-        def categories = inventoryItems?.collect { it.product.category }?.unique()
-        def quantityMap = productAvailabilityService.getQuantityOnHandByInventoryItem(location)
-        def expiredStockMap = [:]
+    def listExpiredStock(InventoryReportCommand command) {
+        command.location = Location.get(session.warehouse.id)
+        List<InventoryItem> inventoryItems = dashboardService.getExpiredStock(command.category, command.location, command.startDate, command.endDate)
+        List<Category> categories = inventoryItems?.collect { it.product.category }?.unique()
+        Map<InventoryItem, Integer> quantityMap = productAvailabilityService.getQuantityOnHandByInventoryItem(command.location)
+        LinkedHashMap<InventoryItem, Integer> expiredStockMap = [:]
```

Comment:
`LinkedHashMap` is an implementation of `Map` interface. 
We should use interfaces as types if possible, hence `Map` should be the type here.
The same case as with Lists - you shouldn't define the type as `ArrayList<String> list = [...]`, but as `List<String> list = [...]`